src/components/cluster/detail/scale_cluster_modal.js

```diff
@@ -88,38 +93,68 @@ class ScaleClusterModal extends React.Component {
         maxValid: nodeCountSelector.scaling.maxValid,
       },
     });
-  };
 
   submit = () => {
     this.setState(
       {
         loading: true,
       },
       () => {
-        var scaling = {
-          min: this.state.scaling.min,
-          max: this.state.scaling.max,
-        };
-
-        this.props.clusterActions
-          .clusterPatch(this.props.cluster, { scaling: scaling })
-          .then(patchedCluster => {
-            this.close();
-
-            new FlashMessage(
-              'The cluster will be scaled within the next couple of minutes.',
-              messageType.SUCCESS,
-              messageTTL.SHORT
-            );
-
-            this.props.clusterActions.clusterLoadDetailsSuccess(patchedCluster);
-          })
-          .catch(error => {
-            this.setState({
-              loading: false,
-              error: error,
+        if (this.state.nodePool) {
```

Comment:
I'm worried about these kinds of conditionals going throughout the UI.

I'd rather have an action called like 'nodePoolorClusterPatch' which takes three parameters (clusterId, Nodepool, scaling) and where nodepool can be null or undefined or something, and then the conditional to decide which api method to call would live in there?

Or perhaps we should make a dedicated `scale_nodepool_modal` instead?

Reply:
As I said thank you for not letting me do this kind of things, but I am rethinking it: one of my main concerns now is get things done here, don't wanna be a blocker in the NPs roadmap, and taking into account that no matter the solution we choose now, we will have to refactor it in the future.... ¿What if we leave this conditional here and refactor it when all clusters are NPs clusters?

Despite everything we discussed, one thing I like about this solution is that is very clear what's going on and shouldn't be too difficult to modify it when needed.

WDYT?

Reply:
What about making a second modal just for scaling node pools?

Then possibly extracting what they have in common if you can find the time?

I'm just concerned that it becomes easy to add another thing, and another thing to this modal, and soon it becomes difficult to make sure cluster scaling works AND node pool scaling works.

Also means we have to test cluster scaling each time we make a change related to node pool scaling.

Reply:
ok, I'm going to try to do it

Reply:
done ping @oponder